### ISSUE TITLE
Add support for Toeplitz*Matrix

### ISSUE THREAD

**Issue by dlfivefifty** [andreasnoack/ToeplitzMatrices.jl#4]
The following gives an unexpected error:
```julia
    T=TriangularToeplitz(rand(10),:U)
    T*rand(10,5)
```
A quick fix would be to do it column-by-column, but I imagine it would be quicker to apply the FFTs directly to matrices.

**Comment by andreasnoack** [comment]
Maybe I should remove the `tmp` array from the type and then just allocate it when doing the multiplication. At that time, I'd know the required size of the temporary complex array. I'll try that out.

**Comment by dlfivefifty** [comment]
A good reason to remove tmp is threadsafeness, right?  And the allocation of a tmp array is a very small fraction of time compared to large FFTs.

**Comment by andreasnoack** [comment]
I spent some time on this yesterday. It's been a while since I looked into the package. FFT plans only works for arrays of a fixed size so either we create a plan for each multiplication/solve and throw it away or we save a vector plan in the Toeplitz matrix and apply that to the columns. I did a few timings and the latter choice seemed to be the fastest.

I don't think that the `tmp` array here will cause a thread safety issue because the array is unique to each matrix instance.

I'll prepare a fix.
